src/components/map/use-map-tariff-zones.ts

```diff
@@ -0,0 +1,112 @@
+import { useCallback, useEffect, useState } from 'react';
+import { Language, useTranslation } from '@atb/translations';
+import mapboxgl from 'mapbox-gl';
+import { getReferenceDataName } from '@atb/utils/reference-data';
+import { centroid } from '@turf/turf';
+import { type TariffZone, getTariffZones } from '@atb/modules/firebase';
+
+export const useMapTariffZones = async (
+  mapRef: React.MutableRefObject<mapboxgl.Map | undefined>,
+) => {
+  const { language } = useTranslation();
+  const [tariffZones, setTariffZones] = useState<TariffZone[]>([]);
+
+  useEffect(() => {
+    const fetchTariffZones = async () => {
+      const zones = await getTariffZones();
+      setTariffZones(zones);
+    };
+
+    fetchTariffZones();
+  }, []);
```

Comment:
I guess this will be replaced at a later point with e.g. `useSWR`? Even though the caching might not work as we want it to with it, I think it's cleaner to use it instead of having a state and a `useEffect`. Should probably confirm whether it's unintended to use `useSWR` for non-`fetch`ing stuff too.